File: app/schemas/perfil.py
"""Pydantic schemas for Perfil entity."""

from datetime import datetime

from pydantic import BaseModel, Field


class PerfilBase(BaseModel):
    nome: str = Field(..., max_length=80)
    descricao: str | None = None


class PerfilCreate(PerfilBase):
    pass


class PerfilUpdate(BaseModel):
    nome: str | None = Field(None, max_length=80)
    descricao: str | None = None


class PerfilResponse(PerfilBase):
    id: int
    ativo: bool
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}

File: frontend/src/types/perfil.ts
/** TypeScript interfaces for Perfil entity matching Pydantic schemas. */

export interface PerfilResponse {
  id: number;
  nome: string;
  descricao: string | null;
  ativo: boolean;
  created_at: string;
  updated_at: string;
}

export interface PerfilCreate {
  nome: string;
  descricao?: string | null;
}

export interface PerfilUpdate {
  nome?: string | null;
  descricao?: string | null;
}
